SWENT-GRP5/SOLVIT
# Manage Interaction between provider and seeker

**Description**

This PR introduces a complete implementation of the interaction between seekers and providers. The key features include:


- _Discussion Button:_ Once a request is assigned, a "Discuss" button is available. Clicking on it allows the seeker to send a message to the provider.
<img width="236" alt="Capture d’écran 2024-11-27 à 20 13 08" src="https://github.com/user-attachments/assets/1674ee50-b692-4c0d-8261-5eda93e39ae8">

<img width="236" alt="Capture d’écran 2024-11-27 à 20 12 48" src="https://github.com/user-attachments/assets/3698da2c-d0e9-472d-b181-2cdc1093dd59">


- _Inbox Functionality:_Seekers and providers can view all messages sent and received in their respective inboxes. The time of the last message sent or received is displayed for convenience.
<img width="236" alt="Capture d’écran 2024-11-27 à 20 12 40" src="https://github.com/user-attachments/assets/1b3995e1-3838-4891-bc04-118a8b7ffa7b">

**Fixes and Improvements**

- Removed hardcoded values for providers and seekers.
- Implemented proper retrieval of user data from Firebase.

**Known Issues and Discussions**

_Legacy Accounts:_

Some accounts created during the first week of the project lack necessary fields in the database. For instance, attempting to view a conversation for such a provider in the inbox will result in no conversation being displayed.
This was implemented with the assumption that we would reset our database to ensure all accounts are correctly initialized.

_Google Login Accounts:_

Accounts created via Google login sometimes lack complete information on Firebase (e.g., missing the "name" field). This can lead to errors when retrieving data.
We should create an issue to address this problem.

**Testing Instructions**

1. Create Test Accounts:

- Manually create two new accounts, one for a provider and one for a seeker, using email registration instead of Google for now.

2. Test Workflow:

- Log in as the seeker and find the provider you created.
- Assign a request to the provider and click "Discuss."
- Send a message to the provider.
- Go to the inbox and verify the message appears there.
- Log out of the seeker account and log in to the provider account.
- Verify that the message from the seeker is displayed and send a response back.
- Confirm that the actions are synchronous by using two phones during the next demo to validate real-time communication.

**Next Steps**

- Reset the database to ensure all accounts are initialized with the necessary fields.
- Address the issue with incomplete data for Google login accounts.
